ExtraDry/ExtraDry.Server/ExtensionMethods/QueryableExtensions.cs

```diff
@@ -159,12 +166,22 @@ internal static IQueryable<T> Sort<T>(this IQueryable<T> source, string? sort, s
             throw new DryException(MissingStabilizerErrorMessage, SortErrorUserMessage);
         }
         if(!string.IsNullOrWhiteSpace(actualSort)) {
-            query = ascending ? 
-                query.OrderBy(sortProperty).ThenBy(modelDescription.StabilizerProperty.ExternalName) : 
-                query.OrderByDescending(sortProperty).ThenByDescending(modelDescription.StabilizerProperty.ExternalName);
+            // If set to always add key, we add a secondary sort to stabilize the sort
+            if(SortStabilization == SortStabilization.AlwaysAddKey) {
+                query = ascending ?
+                    query.OrderBy(sortProperty).ThenBy(modelDescription.StabilizerProperty.ExternalName) :
+                    query.OrderByDescending(sortProperty).ThenByDescending(modelDescription.StabilizerProperty.ExternalName);
+            } else {
+                query = ascending ?
+                    query.OrderBy(sortProperty) :
+                    query.OrderByDescending(sortProperty);
+            }
         }
         else {
-            query = query.OrderBy(modelDescription.StabilizerProperty.ExternalName);
+            // For anything other than ProviderDefaultsOnly, add a default sort for stabilization.
+            if(SortStabilization != SortStabilization.ProviderDefaultsOnly) {
```

Comment:
Shouldn't this also apply for AlwaysAddKey?

Reply:
It does, that's a `!=` but given you read it that way, I'll change it to positive rather than negative